### ISSUE TITLE
[Feature Request]: Allow user defined thumbnail images for files

### ISSUE THREAD

**Issue by ZeaTheEnby** [TagStudioDev/TagStudio#382]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/doc/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

The problem is when it comes to file types that do not have easy thumbnail ability it may be useful to allow the user to pic an image they find useful to distinguish the file. Example a PDF or similiar document that has a useless first page could instead have the image a user decides is the cover photo of the document that displays instead of either a blank symbol representing PDF or a preview of the PDF.

### Solution

My thought on the solution is to just add a field that allows users to input a image that they stored with in a special hidden folder which overrides what ever the thumbnail image is supposed to be.

### Alternatives

Another thing that would at least make this feature less necessary and most likely be easier to implement is simple having the name of the file underneath the thumbnail. This could just be an additional standalone feature (maybe one that can be toggled for people who don't want to see the file names) but i think would add a lot of usability for viewing files other then images. Maybe the Title Field can replace the file name or optionally replace it aswell

**Comment by CyanVoxel** [comment]
I agree, setting a custom thumbnail override would be useful in several circumstances! This was talked about in the Discord server at some point but looks like hasn't had an official feature request until now.

As for the filename underneath the thumbnail, this is planned as an option and is currently tracked via #85

**Comment by kurisocat** [comment]
Custom thumbnails is the one feature preventing my use-case. Would love to see this implemented in the future and it'll surpass any alternative software. Subscribed to the thread. 

P.S. great job on the project so far Cyan it's great software useful to so many friends and their needs.
